src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs

```diff
@@ -56,42 +56,46 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         private async Task MakeFieldReadonlyAsync(
             Document document, SyntaxEditor editor, List<TSymbolSyntax> declarators, CancellationToken cancellationToken)
         {
+            var generator = editor.Generator;
             var declaratorsByField = declarators.GroupBy(g => g.FirstAncestorOrSelf<TFieldDeclarationSyntax>()!);
 
             foreach (var fieldDeclarators in declaratorsByField)
             {
-                var declarationDeclarators = GetVariableDeclarators(fieldDeclarators.Key);
+                var fieldDeclaration = fieldDeclarators.Key;
+                var declarationDeclarators = GetVariableDeclarators(fieldDeclaration);
 
                 if (declarationDeclarators.Count == fieldDeclarators.Count())
                 {
-                    var modifiers = WithReadOnly(editor.Generator.GetModifiers(fieldDeclarators.Key));
-
-                    editor.SetModifiers(fieldDeclarators.Key, modifiers);
+                    var modifiers = WithReadOnly(editor.Generator.GetModifiers(fieldDeclaration));
+                    editor.ReplaceNode(
```

Comment:
It's unclear to me why `editor.SetModifiers` doesn't preserve the trivia

Reply:
SetModifiers actually adds all the annotations to format the node.  i tried removing that, but it broke a lot of other features that were depending on that behavior.  So i spot fixed this way.